### PR TITLE
Simplify `LightningOptimizer`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

We are already copying all these methods into the `__dict__`.

Sister discussion in lite: https://github.com/PyTorchLightning/pytorch-lightning/pull/10175#discussion_r738430790

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/core/optimizer.py [hunk 1]
```diff
@@ -185,10 +162,3 @@ def closure_dis():
         assert trainer is not None
         with trainer.profiler.profile(profiler_action):
             trainer.accelerator.optimizer_step(self._optimizer, self._optimizer_idx, closure, **kwargs)
-
-    def __repr__(self) -> str:
```
[the remark this answers is not in the document]
Comment (tchaton) [reply pytorch_lightning/core/optimizer.py:189]: I think it is still useful to know you aren't using our own optimizer, but the lightning optimizer one. This is definitely useful for debugging purposes.

[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/core/optimizer.py:189]: The test below shows it. 
You can try:

```py
    opt = LightningOptimizer(torch.optim.Adam(torch.nn.Linear(1, 1).parameters()))
    print(opt)
```
    
prints on master:
```py
LightningAdam(groups=[{'amsgrad': False, 'betas': (0.9, 0.999), 'eps': 1e-08, 'lr': 0.001, 'weight_decay': 0}])
```

prints here:
```
LightningAdam (
Parameter Group 0
    amsgrad: False
    betas: (0.9, 0.999)
    eps: 1e-08
    lr: 0.001
    weight_decay: 0
)
```

And if you did

```py
    opt = torch.optim.Adam(torch.nn.Linear(1, 1).parameters())
    print(opt)
```

you get 

```
Adam (
Parameter Group 0
    amsgrad: False
    betas: (0.9, 0.999)
    eps: 1e-08
    lr: 0.001
    weight_decay: 0
)
```

[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/core/optimizer.py:189]: I removed it because this is supposed to be a tiny wrapper for us to hook into our accelerators.

By writing our own `__repr__`, we override the one available in the optimizer passed.

Perhaps the user has its own custom Optimizer, with its own custom `__repr__`. We want to use it in that case.

If you think that the PyTorch optimizers should have a more readable `__repr__`, this is something that should be raised and implemented in PyTorch directly. This wrapper should not make choices for the user.

[the remark this answers is not in the document]
Comment (carmocca) [reply pytorch_lightning/core/optimizer.py:189]: Opened https://github.com/pytorch/pytorch/issues/67587.

Resolving this
